TopMovies/View/TopMoviesVC/TopMoviesViewController.swift

```diff
@@ -11,6 +11,10 @@ import SnapKit
 // MARK: - Props struct
 struct TopMoviesProps {
   let movieCategories: [MovieCategoryProps]
+  
+  func allButtonClicked(_ categoryId: MovieCategory.ID) {
+    Router.shared.perform(route: .categoryVC(categoryId: categoryId))
+  }
```

Comment:
I'd prefer closures instead of a function, since you may have additional logic inside of connector and you may require current state for that. So it has to be as a closure:
`let allButtonClickedAction: () -> Void`
Moreover, that action relates to `MovieCategoryProps`, since it's an action, that performs route. This way you will be able to customize those routes if each category opens different screens

id for those props should be stored inside of that struct
`let id: MovieCategory.ID`